### ISSUE TITLE
Incorrectly parsing escape char in raw string (eg: `\{`)

### ISSUE THREAD

**Issue by rodrigozhou** [Instagram/LibCST#462]
LibCST v0.3.17 is raising an exception when parsing this valid code:

```
a = "'"
print(fr"\{a}")
```

Stack trace:
```
Traceback (most recent call last):
  File ".../libcst/_parser/base_parser.py", line 152, in _add_token
    plan = stack[-1].dfa.transitions[transition]
KeyError: ReservedString(})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../libcst/tool.py", line 832, in <module>
    main(os.environ.get("LIBCST_TOOL_COMMAND_NAME", "libcst.tool"), sys.argv[1:])
  File ".../libcst/tool.py", line 827, in main
    return lookup.get(args.action or None, _invalid_command)(proc_name, command_args)
  File ".../libcst/tool.py", line 272, in _print_tree_impl
    tree = parse_module(
  File ".../libcst/_parser/entrypoints.py", line 71, in parse_module
    result = _parse(
  File ".../libcst/_parser/entrypoints.py", line 51, in _parse
    result = parser.parse()
  File ".../libcst/_parser/base_parser.py", line 111, in parse
    self._add_token(token)
  File ".../libcst/_parser/base_parser.py", line 187, in _add_token
    raise ParserSyntaxError(
libcst._exceptions.ParserSyntaxError: Syntax Error @ 29:17.
Incomplete input. Encountered '}', but expected 'FSTRING_END', 'FSTRING_STRING', or '{'.
```

My guess is LibCST is escaping `"\{"`, but it wasn't supposed to escape since it's a raw string.

**Comment by thatch** [comment]
Bisect says I introduced this in 02fc4401bc3313f0db109d86d1b84c6b35abba5a which probably means it's broken in parso as well.  See https://github.com/Instagram/LibCST/pull/466

**Comment by zsol** [comment]
This should be fixed by #466.
